aws/nss_aws_login

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+#
+# Convenience script for logging out then logging in also
+# getting temporary AWS credentials from the command line
+# and setting them to be the 'default' profile.
+# https://aws.amazon.com/premiumsupport/knowledge-center/sso-temporary-credentials/
+#
+# IMPORTANT NOTES:
+# - This assumes that the first profile in your AWS credentials file is the
+# 'default' one, and will make updates to it accordingly. Specifically, it will
+# overwrite lines 2 (key_id), 3 (secret), and 4 (token).
+# - If you have multiple profiles in the file, make sure they don't start
+# until line 5.
+# - If the credentials file exists, it will make a single backup copy. If
+# the credentials file does NOT exist, it will create an empty one.
+#
+
+profile_name="$1"
+if [ -z "$profile_name" ]; then
+  echo "usage: $(basename $0) PROFILE_NAME"
+  exit 1
+fi
+
+if ! aws configure list-profiles | grep $profile_name >/dev/null; then
+  echo "ERROR: Invalid AWS PROFILE_NAME specified"
+  exit 1
+fi
+
+echo "## Performing AWS Login with the profile $profile_name."
+echo "##"
+
+echo "## Step 1: aws sso logout"
+aws sso logout
+
+echo "## Step 2: aws sso login --profile $profile_name"
+aws sso login --profile $profile_name
+
+echo
+echo "## Step 3: make $profile_name the default profile"
+
+ssh_cache_file=$(find ~/.aws/sso/cache -maxdepth 1 -type f -name "[0-9]*.json" -exec grep -l accessToken {} \;)
+if [ -z "$ssh_cache_file" ]; then
+  echo "ERROR: Cannot find SSO file - have you run 'aws sso login --profile $profile_name' first?"
+  exit 1
+fi
+
+CREDS=~/.aws/credentials
+if [ ! -e $CREDS ]; then
+  echo "## Creating an empty credentials file"
+  mkdir -p $(dirname $CREDS)
+  touch $CREDS
+  chmod 600 $CREDS
+  echo "[default]" > $CREDS
+  echo "aws_access_key_id=tmp" >> $CREDS
+  echo "aws_secret_access_key=tmp" >> $CREDS
+  echo "aws_session_token=tmp" >> $CREDS
+else
+  echo "## Making backup copy of existing credentials file"
+  cp $CREDS ${CREDS}.bak
+fi
+# set -xv   # this line will enable debug
+echo "## Getting Account and Role Name from config"
+account_id=$(aws configure get sso_account_id --profile $profile_name)
+role_name=$(aws configure get sso_role_name --profile $profile_name)
+region=$(aws configure get region --profile $profile_name)
+
+# set the default region in config
+aws configure set region $region
+
+token=$(cat $ssh_cache_file | jq -r '.accessToken')
+
+# if credentials have expired (e.g. overnight) this commmand will fail, thus the check here
+echo "## Getting SSO role credentials"
+
+credentials=$(aws sso get-role-credentials --role-name $role_name --account-id $account_id --access-token $token --region $region)
```

Comment:
pass the region

Reply:
do we need to specify the region if the default region has been configured?

Reply:
We do not, I'll remove